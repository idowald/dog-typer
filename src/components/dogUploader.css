#upload-photo {
  display: none;
}
.dogUploader{
  margin-left: 255px;
}
