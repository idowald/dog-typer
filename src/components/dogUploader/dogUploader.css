#upload-photo {
  display: none;
}
.dogUploader{
  width: 180px;
}
