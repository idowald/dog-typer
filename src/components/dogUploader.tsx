import * as React from "react";
import {
  Button,
  CircularProgress,
  createStyles,
  Theme
} from "@material-ui/core";
import "./dogUploader.css";
import { makeStyles } from "@material-ui/core/styles";

export interface DispatchProps {
  // TODO, problem of versions between ActionCreator and Action, needs fixing https://github.com/reduxjs/redux/issues/2709
  changeDog: any;
}

export interface OwnProps {
  loadingModel: boolean;
}
const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    hidden: {
      visibility: "hidden"
    }
  })
);
export const DogUploader = ({
  changeDog,
  loadingModel
}: DispatchProps & OwnProps) => {
  const classes = useStyles();
  return (
    <label htmlFor="upload-photo" className="dogUploader">
      <Button
        color="primary"
        variant="contained"
        component="span"
        onChange={changeDog}
        disabled={loadingModel}
      >
        Upload your dog
        <CircularProgress
          color="inherit"
          className={loadingModel ? "" : classes.hidden}
        />
        <input
          id="upload-photo"
          name="upload-photo"
          type="file"
          accept="image/*"
        />
      </Button>
    </label>
  );
};
