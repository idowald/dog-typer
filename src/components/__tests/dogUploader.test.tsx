import React from "react";
import renderer from "react-test-renderer";

import {DogUploader} from "../dogUploader/dogUploader";

test("renders learn react link", () => {
  const rendered = renderer.create(<DogUploader changeDog={()=>{}} loadingModel={false} />);
  expect(rendered.toJSON()).toMatchSnapshot();
});
