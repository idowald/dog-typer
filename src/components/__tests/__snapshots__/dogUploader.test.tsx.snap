// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dog uploader 1`] = `
<label
  className="dogUploader"
  htmlFor="upload-photo"
>
  <span
    aria-disabled={false}
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    onBlur={[Function]}
    onChange={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      className="MuiButton-label"
    >
      Upload your dog
      <div
        className="MuiCircularProgress-root makeStyles-hidden-1 MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </div>
      <input
        accept="image/*"
        id="upload-photo"
        name="upload-photo"
        type="file"
      />
    </span>
  </span>
</label>
`;
